[Bug] multi-user app crash

### Steps to reproduce/复现步骤

1.  adb shell create-user 10 
1.  adb shell pm list users
```
Users:
        UserInfo{0:机主:13} running
        UserInfo{10:qb:30} running
        UserInfo{11:bin:30} running
        UserInfo{12:quan:30} running
```


### Expected behaviour/预期行为

launcher app

### Actual behaviour/实际行为

app crash 

### Xposed Module List/Xposed 模块列表

```Shell
no add Module
```


### LSPatch version/LSPatch 版本

0.5.1

### Android version/Android 版本

HomeOS 2.0.0

### Shizuku version/Shizuku 版本

13.3.0.r993.5392b9d

### Version requirement/版本要求

- [X] I am using latest debug CI version of LSPatch and enable verbose log/我正在使用最新 CI 调试版本且启用详细日志

### Apk file/Apk 文件

com.tencent.mm 8-0-33

### Logs/日志

<details>

```

2023-07-29 16:44:08.420 27147-27147/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.tencent.mm, PID: 27147
    java.lang.ExceptionInInitializerError
        at org.lsposed.lspatch.metaloader.LSPAppComponentFactoryStub.<clinit>(Unknown Source:420)
        at java.lang.Class.newInstance(Native Method)
        at android.app.LoadedApk.createAppFactory(LoadedApk.java:270)
        at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:1005)
        at android.app.LoadedApk.getClassLoader(LoadedApk.java:1129)
        at android.app.LoadedApk.getResources(LoadedApk.java:1419)
        at android.app.ContextImpl.createAppContext(ContextImpl.java:2662)
        at android.app.ContextImpl.createAppContext(ContextImpl.java:2644)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7452)
        at android.app.ActivityThread.access$2600(ActivityThread.java:260)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2435)
        at android.os.Handler.dispatchMessage(Handler.java:110)
        at android.os.Looper.loop(Looper.java:219)
        at android.app.ActivityThread.main(ActivityThread.java:8668)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)
     Caused by: java.lang.SecurityException: get application info: Neither user 1010247 nor current process has android.permission.INTERACT_ACROSS_USERS.
        at android.os.Parcel.createException(Parcel.java:2091)
        at android.os.Parcel.readException(Parcel.java:2059)
        at android.os.Parcel.readException(Parcel.java:2007)
        at android.content.pm.IPackageManager$Stub$Proxy.getApplicationInfo(IPackageManager.java:5211)
        at org.lsposed.lspatch.metaloader.LSPAppComponentFactoryStub.<clinit>(Unknown Source:182)
        at java.lang.Class.newInstance(Native Method) 
        at android.app.LoadedApk.createAppFactory(LoadedApk.java:270) 
        at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:1005) 
        at android.app.LoadedApk.getClassLoader(LoadedApk.java:1129) 
        at android.app.LoadedApk.getResources(LoadedApk.java:1419) 
        at android.app.ContextImpl.createAppContext(ContextImpl.java:2662) 
        at android.app.ContextImpl.createAppContext(ContextImpl.java:2644) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7452) 
        at android.app.ActivityThread.access$2600(ActivityThread.java:260) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2435) 
        at android.os.Handler.dispatchMessage(Handler.java:110) 
        at android.os.Looper.loop(Looper.java:219) 
        at android.app.ActivityThread.main(ActivityThread.java:8668) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109) 
     Caused by: android.os.RemoteException: Remote stack trace:
        at android.app.ContextImpl.enforce(libmapleframework.so:5913644)
        at android.app.ContextImpl.enforceCallingOrSelfPermission(libmapleframework.so:5911004)
        at com.android.server.pm.permission.PermissionManagerService.enforceCrossUserPermission(libmapleservices.so:7798444)
        at com.android.server.pm.PackageManagerService.getApplicationInfoInternal(libmapleservices.so:7633196)
        at com.android.server.pm.PackageManagerService.getApplicationInfo(libmapleservices.so:7632916)
    callee: null 1615/2990



```

![image](https://github.com/LSPosed/LSPatch/assets/10138880/3688ae1e-fdd8-4e04-9212-512f7c710829)

</details>

--- a/meta-loader/src/main/java/org/lsposed/lspatch/metaloader/LSPAppComponentFactoryStub.java
+++ b/meta-loader/src/main/java/org/lsposed/lspatch/metaloader/LSPAppComponentFactoryStub.java
@@ -5,6 +5,7 @@ import android.app.AppComponentFactory;
 import android.content.pm.ApplicationInfo;
 import android.content.pm.IPackageManager;
 import android.os.Build;
+import android.os.Process;
 import android.os.ServiceManager;
 import android.util.JsonReader;
 import android.util.Log;
@@ -70,9 +71,9 @@ public class LSPAppComponentFactoryStub extends AppComponentFactory {
                 var ipm = IPackageManager.Stub.asInterface(ServiceManager.getService("package"));
                 ApplicationInfo manager;
                 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-                    manager = (ApplicationInfo) HiddenApiBypass.invoke(IPackageManager.class, ipm, "getApplicationInfo", Constants.MANAGER_PACKAGE_NAME, 0L, 0);
+                    manager = (ApplicationInfo) HiddenApiBypass.invoke(IPackageManager.class, ipm, "getApplicationInfo", Constants.MANAGER_PACKAGE_NAME, 0L, Process.myPid() / 100000);
                 } else {
-                    manager = ipm.getApplicationInfo(Constants.MANAGER_PACKAGE_NAME, 0, 0);
+                    manager = ipm.getApplicationInfo(Constants.MANAGER_PACKAGE_NAME, 0, Process.myUid() / 100000);
                 }
                 try (var zip = new ZipFile(new File(manager.sourceDir));
                      var is = zip.getInputStream(zip.getEntry(Constants.LOADER_DEX_ASSET_PATH));

